src/client/app/containers/import.ts

```diff
@@ -213,4 +219,36 @@ export class Import {
     cancel() {
         this._store.dispatch(new UI.ToggleImportAction(false));
     }
+
+    checkForViewSnippet() {
+        if (!this.hasViewSettings()) {
+            return;
+        }
+
+        let viewData = Office.context.document.settings.get('VIEW_URL_SETTINGS');
+        if (viewData.type === 'samples') {
+            let hostJsonFile = `${environment.current.config.samplesUrl}/view/${environment.current.host.toLowerCase()}.json`;
+            let sub = this._request.get<JSON>(hostJsonFile, ResponseTypes.JSON)
+                .subscribe(lookupTable => {
+                    if (lookupTable && lookupTable[viewData.id]) {
+                        this._store.dispatch(new Snippet.ImportAction(Snippet.ImportType.SAMPLE, lookupTable[viewData.id]));
+                    }
+
+                    if (sub && !sub.closed) {
+                        sub.unsubscribe();
+                    }
+                });
+
+        } else {
+            // Even though user is in editor mode, dispatch with flag to avoid saving gist until user begins typing
+            this._store.dispatch(new Snippet.ImportAction(Snippet.ImportType.GIST, viewData.id, true /*isViewMode*/));
```

Comment:
In the context of an Import, being in "view mode" means not saving the snippet to local storage, regardless of whether it's a gist or sample. So here, even though this is all editor mode code, the flag is still there since we don't want to save until the user types.